service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/utils/TaskStatusChecker.java

```diff
@@ -49,33 +56,25 @@ public static synchronized void init(CassandraConnectionProvider cassandraConnec
 
 
     public boolean hasKillFlag(long taskId) {
-        CacheItem cacheItem = getCacheItem(taskId);
-        if (!cacheItem.hasKilledFlag && cacheItem.lastCheck + CHECKING_INTERVAL < System.currentTimeMillis()) {
-            synchronized (instance) {
-                if (cacheItem.lastCheck + CHECKING_INTERVAL < System.currentTimeMillis()) {
-                    LOGGER.info("Checking the cancellation from the backend database for task id {}", taskId);
-                    cacheItem.lastCheck = System.currentTimeMillis();
-                    if (taskDAO.hasKillFlag(taskId)) {
-                        cacheItem.hasKilledFlag = true;
-                        cache.put(taskId, cacheItem);
-                    }
-                }
-            }
-
+        try {
+            return cache.get(taskId);
+        } catch (ExecutionException e) {
+            LOGGER.info(e.getMessage());
+            return false;
         }
-        return cacheItem.hasKilledFlag;
     }
 
-
-    private CacheItem getCacheItem(long taskId) {
-        ConcurrentMap<Long, CacheItem> map = cache.asMap();
-        map.putIfAbsent(taskId, new CacheItem());
-        return map.get(taskId);
-    }
-
-    private static class CacheItem {
-        volatile boolean hasKilledFlag = false;
-        volatile long lastCheck = 0;
+    /*
```

Comment:
Why not proper javadoc instead?

Reply:
This is a private method. And as a rule: your code should be clear and expressed enough so anyone can understand it.
Adding comments to describe what the code is doing is always a code smell.
This comment is the exception which proves the rule and it was added to notify the code reader about the capability of the com.google.common.cache library. In case he/she didn't read its documentation before.

Reply:
I agree, but not always.. Sometimes it's useful to have javadoc as well, even though the code is as clear as possible. In this case thought that javadoc would be "nicer" instead of the already existent block comment.